### DISCUSSION TITLE
"abbreviated" file names

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#150] [Ideas]
it is a general problem that Medley uses full pathnames in LOADEDFILELST and other places that ties a file to the local naming conventions.

I'm thinking of a kind of "abbreviation" device so that {MEDLEY}/lispusers could/would be used instead of {DSK}<users>larry>ilisp>medley>lispusers>  everywhere  -- it would need to tie into the many different paths for file-name hacking to return {medley}blah instead of {DSK}. (Or maybe just never traffic in {DSK} files and always make them relative to {MEDLEY} or {MAIKO} or {HOME} or {LOGIN} ?
The DSK.c and DIR.c code is pretty opaque. I don't like making this automatic for all environment variables (or ones ending in ... DIR.
One convention I've seen is the idea of using punctuation {$MEDLEY} means look up MEDLEYDIR.

**Comment by nbriggs** [comment]
Yes, but it can be important to know exactly which file it was that was loaded into the sysout.  Did I load `XXX.DFASL` from `{dsk}<Users>briggs>lisp>` ? or was it the version in `{dsk}<usr>local>medley>` ?

**Reply by masinter** [reply]
the problem is that the only thing that the file-name-at loadup only has limited significance... of the components
[ device/host/path to medleydir]  [where in medleydir] [file name] . [file extension] ; [version]

* The file name is significant.
* The file extension should tell you which compiler was used (or that it was a lisp source)
* the version number at the time of loadup doesn't tell you anything
* "Where in medleydir" doesn't tell you much -- pretty much every file name has a unique place
* the device/host/path gives some clues about who did the loadup maybe
